codebase/analysis/src.lib/igrf_v2/igrf.1.0/src/igrflib.c

```diff
@@ -1532,3 +1840,72 @@ int AACGM_v2_RK45(double xyz[], int idir, double *ds, double eps, int code) {
   return (0);
 }
 
+/*-----------------------------------------------------------------------------
+;
+; NAME:
+;       ecdip_mlt_ref (translated from Pierre-Louis Blelly Matlab code)
+;
+; PURPOSE:
+;       Determine reference MLT location for given date/time. Used for
+;       eccentric dipole coordinates.
+;
+; CALLING SEQUENCE:
+;       mlt_ref = ecdip_mlt_ref(yr,mo,dy,hr,mt,sc);
+;
+;     Input Arguments:
+;       yr,mo,dy      - Date
+;       hr,mt,sc      - Time
+;
+;     Return Value:
+;       mlt           - mlt reference of given date/time
+;
+;+-----------------------------------------------------------------------------
+*/
+double ecdip_mlt_ref(int yr, int mo, int dy, int hr, int mt, int sc)
+{
+  int i,k;
+  double lonmag_ref, tilt;
+  double t,gst,M,L,R,obliq,slp;
+  double cgst,sgst,sob,cob,cslp,sslp;
+  double fday, julian;
+  double pos[3], coord[3];
+
+  fday = (hr + mt/60. + sc/3600.)/24.;
+  julian = TimeYMDHMSToJulian(yr,mo,dy,hr,mt,sc) -
+           TimeYMDHMSToJulian(1899,12,31,12,0,0);
+
+  lonmag_ref = tilt = 0.;
+
+  t     = julian/36525.;
+  L     = MOD(279.696678 + 0.9856473354*julian, 360);
+  gst   = MOD(279.690983 + 0.9856473354*julian + 360.*fday + 180., 360);
+  M     = MOD(358.475845 + 0.985600267*julian, 360);
+  L    += (1.91946 - 0.004789*t)*sin(M*DTOR) + 0.020094*sin(2*M*DTOR);
+  obliq =  23.45229 - 0.0130125*t;
+  slp   = L - (0.0055686 - 0.025e-4*t);
+
+  cgst = cos(gst*DTOR);
+  sgst = sin(gst*DTOR);
+  sob  = sin(obliq*DTOR);
+  cob  = cos(obliq*DTOR);
+  sslp = sin(slp*DTOR);
+  cslp = cos(slp*DTOR);
+
+  pos[0] = -( cslp*cgst + sslp*sgst*cob) * RE;
+  pos[1] = -(-cslp*sgst + sslp*cgst*cob) * RE;
+  pos[2] = -( sslp*sob ) * RE;
+
+  for (k=0; k<3; k++) pos[k] -= ecdip.pos[k];
+
+  for (k=0; k<3; k++) {
+    coord[k] = 0.;
+    for (i=0; i<3; i++) coord[k] += ecdip.g2m[k][i]*pos[i];
+  }
+
+  lonmag_ref = MOD(360 + atan2(coord[1],coord[0])/DTOR, 360);
+  R          = sqrt(coord[0]*coord[0] + coord[1]*coord[1] + coord[2]*coord[2]);
+  tilt       = -asin(coord[2]/R)/DTOR;
```

Comment:
This variable is not used, is it meant to be returned?  Should it exist in a separate function?